.evergreen/lib/server_setup.rb

```diff
@@ -0,0 +1,93 @@
+require 'mongo'
+
+Mongo::Logger.logger.level = :WARN
+
+class ServerSetup
+  def setup_aws_auth
+=begin
+    require_env_vars(%w(
+      MONGO_RUBY_DRIVER_AWS_AUTH_ACCESS_KEY_ID
+      MONGO_RUBY_DRIVER_AWS_AUTH_SECRET_ACCESS_KEY
+      MONGO_RUBY_DRIVER_AWS_AUTH_USER_ARN
+    ))
+=end
+
+    arn = env!('MONGO_RUBY_DRIVER_AWS_AUTH_USER_ARN')
+    puts "Adding AWS-mapped user #{arn}"
+    create_aws_user(arn)
+
+=begin The credentials to be used here need to be the temporary credentials
+corresponding to the user being added, not credentials of the (administrative)
+user that performed user creation a moment ago. In order for this to happen
+this script must be passed two sets of credentials and the second set must be
+obtained somehow. Instead of doing this we should simply start the server
+initially without auth, add the AWS-mapped user, then restart the server with
+auth such that there is never a bootstrap user to begin with.
+
+    puts "Removing bootstrap user"
+    aws_client = Mongo::Client.new(%w(localhost),
+      user: env!('MONGO_RUBY_DRIVER_AWS_AUTH_ACCESS_KEY_ID'),
+      password: env!('MONGO_RUBY_DRIVER_AWS_AUTH_SECRET_ACCESS_KEY'),
+      auth_mech: :aws,
+      auth_mech_properties: {'aws_session_token' => ENV['MONGO_RUBY_DRIVER_AWS_AUTH_SESSION_TOKEN']},
+    )
+    aws_client.use('admin').database.users.remove('bootstrap')
+    aws_client.close
+
+    puts "Verifying bootstrap user was removed"
+    begin
+      bootstrap_client['test-coll'].insert_one(test: true)
+    rescue Mongo::Error::OperationFailure => e
+      if e.message =~ /command insert requires authentication/
+        # Expected outcome
+      else
+        raise
+      end
+    else
+      raise 'Expected the bootstrap user to had been deleted, but it was not'
+    end
+=end
```

Comment:
Is there a scenario in which this code should be un-commented?

Reply:
Probably not, removed it to https://gist.github.com/p-mongo/2829f9fece71753b40133ae0d180fcb2.